src/BenchmarkDotNet/Helpers/ArtifactFileNameHelper.cs

```diff
@@ -18,7 +19,11 @@ internal static string GetTraceFilePath(DiagnoserActionParameters details, DateT
         {
             string nameNoLimit = GetFilePathNoLimits(details, creationTime, fileExtension);
 
-            int limit = PathFeatures.AreAllLongPathsAvailable() ? CommonSenseLimit : WindowsOldPathLimit;
+            // long paths can be enabled on Windows but it does not mean that ETW is going to work fine..
+            // so we always use 260 as limit on Windows
+            int limit =  RuntimeInformation.IsWindows()
+                ? WindowsOldPathLimit - "userheap.etl".Length // the session files get merged, they need to have same name (without extension)
```

Comment:
You could reuse HeapSession.FileExtension. I checked it seems to be the longest extension, so it is a good solution.

Reply:
> You could reuse HeapSession.FileExtension

I was thinking about it. The problem is that it's an instance property - so I would need to have an `HeapSession` instance to get it.

Then I was thinking about introducing an `internal const string` field to the `HeapSession` but it was far from perfect: 
* I would need two names for the same thing
* it would be still possible to introduce a new session type with a longer name.

I think that we can keep it as it is until we find a better solution.